vignettes/bobRoss.Rmd

```diff
@@ -0,0 +1,181 @@
+
+#rename,switch columns and calculate percentage over all paintings and frequency though all episodes
+per_features <- temp %>% tibble::rownames_to_column() %>% 
+  select(feature=rowname, tot = V1) %>% 
+  mutate(
+    feature_percentage = (tot / sum(tot))*100, 
+    feature_percentage_Label = paste0(round(feature_percentage,1),"%"), 
+    feature_freq = tot/ nrow(df)*100, 
+    feature_freq_label = paste0(round(feature_freq,1),"%"))
+```
+
+```{r feature frequency plot}
+feature_freq_cut <- 10 #10% most present features
+ggplot(data=filter(per_features,feature_freq>feature_freq_cut), aes(x=reorder(feature,feature_freq),y=feature_freq)) + 
+  geom_bar(stat='identity') + geom_text(aes(label=feature_freq_label), position=position_dodge(width=0.9), vjust=.5,hjust=0,size=2.5,color='red') + 
+  coord_flip() + theme_fivethirtyeight() + ggtitle('Features\'s appearance(%) through all episodes')
+```
+
+* `tree` and `trees` features appear in more than 90% of all the paintings.
+
+## Correlation
+Since a row with no entries causes a standard deviation = 0, features are selected based on their number. 
+
+```{r feature correlation}
+#find features present
+top<-c(per_features %>% filter(tot>1) %>% arrange(-tot) %>% select(feature))
+num_data<-df %>% select_(.dots = top$feature)
+num_cols <- sapply(num_data, is.numeric)
+corrplot(cor(num_data[,num_cols]), method='square',order="AOE")
+```
+
+* we see positive correlation for the expected cases, like `tree` / `trees`, or `night` / `moon`
+* we also see negative correlation for features totally different, such as `waves` / `tree` 
+* a negative correlation means that as one of the variables increases, the other tends to decrease, and vice versa, so it makes sense to find an anti-correlation in the case `waves` / `tree` for example.
+
+## Study by Episodes
+### Episodes having the greatest number of features
+```{r episode prep.}
+per_episode <- df %>% 
+  select(-episode,-season,-episode_num ,-title) %>% 
+  select_if(is.numeric) %>% 
+  summarise(sum = rowSums(.)) %>% 
+  cbind(episode = df$title) %>% arrange(-sum)
+
+#select a cut
+cut_features<-11
+ggplot(data=filter(per_episode,sum>cut_features), aes(x=reorder(episode,sum),y=sum)) + 
+  geom_bar(stat='identity') + 
+  coord_flip() + theme_fivethirtyeight() + ggtitle(paste0('Paintings having more than ', cut_features,' features'))
+```
+
+### Episodes distribution vs. their number of features
+```{r episoe plot}
+per_episode_summary <- per_episode %>% 
+  group_by(sum) %>% 
+  dplyr::summarise(tot_features=n()) %>%
+  mutate(
+    percent = (tot_features/ sum(tot_features))*100, 
+    label = paste0(round(percent,1),"%"))
+
+ggplot(data=per_episode_summary, aes(x=sum,y=tot_features)) + 
+  geom_bar(stat='identity') + 
+  geom_text(aes(label=label), position=position_dodge(width=0.9), vjust=-1,hjust=.5,size=2.5,color='red')+
+  theme_fivethirtyeight() + ggtitle('Distribution of paitings vs. \n number of features')
```

Comment:
- Improve title of plot to be more clear, such as " Distribution of number of features per episode"
- label axes of plot for clarity, such as, x-axis = Number of features, y = axis = Number of episodes